src/library.js

```diff
@@ -6264,16 +6536,25 @@ LibraryManager.library = {
   // netinet/in.h
   // ==========================================================================
 
+#if USE_PTHREADS
+  in6addr_any: '; if (ENVIRONMENT_IS_PTHREAD) _in6addr_any = PthreadWorkerInit._in6addr_any; else PthreadWorkerInit._in6addr_any = _in6addr_any = allocate([0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0], "i8", ALLOC_STATIC)',
+  in6addr_loopback: '; if (ENVIRONMENT_IS_PTHREAD) _in6addr_loopback = PthreadWorkerInit._in6addr_loopback; else PthreadWorkerInit._in6addr_loopback = _in6addr_loopback = allocate([0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1], "i8", ALLOC_STATIC)',
+#else
   in6addr_any:
     'allocate([0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0], "i8", ALLOC_STATIC)',
   in6addr_loopback:
     'allocate([0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,1], "i8", ALLOC_STATIC)',
+#endif
 
   // ==========================================================================
   // netdb.h
   // ==========================================================================
 
+#if USE_PTHREADS
```

Comment:
this pattern seems to show up a lot. it seems there is unnecessary duplication in 2 areas: first, the allocate (or whatever the actual creation is) appears in two places, in both sides of the ifdef. second, the check if is pthread, then use it from PthreadWorkerInit, etc.

How about ` __h_errno_state: {{{ pthreadify('allocate(1, "i32", ALLOC_STATIC)') }}}`, where `pthreadify` does all the stuff automatically?